### PR TITLE
[CIS-1262, CIS-1293] Fix message list jumps and cell not resizing on edit

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Link
CIS-1262
CIS-1293

### 🎯 Goal
- Fix message list scrolling jumps when a new message is received.
- Fix message cell size not updated when editing a message with bigger/smaller content.
- **Bonus:**
     - Fix not executing the `completion` of `updateMessages()` which was not being called ever.
     - Removes the list changes conflicts log.

### 🛠 Implementation
- Scrolling jumps fix: Whenever the message list is reloaded, it re-calculates the message cell's height, causing a scrolling jump. When using `performBatchUpdates` this is not an issue. Use `performBatchUpdates` only for inserts. We also need to reload the previous cell to make sure the timestamp is removed if needed.
- Message cell resize fix: When a message is edited, previously we were just updating the message cell content and not reloading the cell. Now, if the text of the message is different, we also reload the cell, to re-calculate the cell size.

### 🧪 Testing
- Scrolling jumps fix: 
    1. Scroll up in the message list
    2. Send a message from another device
    3. Verify that the message list doesn't move
- Message cell resize fix:
   1. Edit a message with more lines of text than it had before
   2. Verify the cell changes its size.

### 🎨 Changes
| Message cell resize fix |  Scrolling jumps fix |
| --- | --- |
| <video src="https://user-images.githubusercontent.com/12814114/141162863-ef1803a2-8da8-4645-9fa6-2eecb120cf61.mp4"> | <video src="https://user-images.githubusercontent.com/12814114/141162941-9f54c514-537e-418f-aa48-e21787fdef8d.mp4"> |

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift [hunk 1]
```diff
@@ -188,8 +191,16 @@ open class ChatMessageListView: UITableView, Customizable, ComponentsProvider {
...
             case let .insert(message, index: index):
                 UIView.performWithoutAnimation {
-                    self.reloadData()
+                    self.performBatchUpdates {
+                        self.insertRows(at: [index], with: .none)
+                    } completion: { _ in
+                        // Update previous row to remove timestamp if needed
+                        // +1 instead of -1 because the message list is inverted
+                        let previousIndex = IndexPath(row: index.row + 1, section: index.section)
+                        self.reloadRows(at: [previousIndex], with: .none)
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift:200]: Is OK for the case when the inserted message is the only message in the channel ie. when the previous cell we're reloading does not exist?

File: Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift [hunk 2]
```diff
@@ -228,11 +239,13 @@ open class ChatMessageListView: UITableView, Customizable, ComponentsProvider {
                 cellBeforeUpdateReuseIdentifier == cellAfterUpdateReuseIdentifier,
                 cellBeforeUpdateMessage?.id == cellAfterUpdateMessage?.id,
                 cellBeforeUpdateMessage?.type == cellAfterUpdateMessage?.type,
-                cellBeforeUpdateMessage?.deletedAt == cellAfterUpdateMessage?.deletedAt {
+                cellBeforeUpdateMessage?.deletedAt == cellAfterUpdateMessage?.deletedAt,
+                cellBeforeUpdateMessage?.text == cellAfterUpdateMessage?.text {
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift:243]: Would be nice to add these scenarios to the pre-release QA doc to make sure we don't have regressions related to cell reloading

**Review by evsaev — CHANGES_REQUESTED** [review]
Let's check the case with a single message and if it's OK I'll approve 👍

File: Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift:201]: Nice catch, again 👏

**Review by evsaev — APPROVED** [review]
🚢

**Pull request merged**